### PR TITLE
CSHARP-2838: Remove redundant AWS test expectation

### PR DESCRIPTION
**Pull request by MikalaiMazurenka**
Evergreen: https://evergreen.mongodb.com/version/5e8283f1c9ec4420a7a316b6

### CODE REVIEW

**Review by rstam — APPROVED** [review]
LGTM

File: tests/MongoDB.Driver.Core.Tests/Core/Authentication/MongoAWSAuthenticatorTests.cs [hunk 1]
```diff
@@ -126,8 +126,6 @@ static MongoAWSAuthenticatorTests()
 
             var actualRequestId0 = sentMessages[0]["requestId"].AsInt32;
             var actualRequestId1 = sentMessages[1]["requestId"].AsInt32;
-            actualRequestId0.Should().Be(expectedRequestId);
```
Comment (DmitryLukyanov) [inline tests/MongoDB.Driver.Core.Tests/Core/Authentication/MongoAWSAuthenticatorTests.cs:129]: Should this line also be removed?

    var expectedRequestId = RequestMessage.CurrentGlobalRequestId + 1;

**Review by DmitryLukyanov — APPROVED** [review]
Some code clearing and then LGTM

File: tests/MongoDB.Driver.Core.Tests/Core/Authentication/MongoAWSAuthenticatorTests.cs (hunk 1 above)
Comment (MikalaiMazurenka) [reply tests/MongoDB.Driver.Core.Tests/Core/Authentication/MongoAWSAuthenticatorTests.cs:129]: Good point.
Done.

**Review by vincentkam — APPROVED** [review]
LGTM! 😸

**Pull request closed without merging**